flang/lib/Lower/TransformationalRuntime.cpp

```diff
@@ -18,6 +19,26 @@
 
 using namespace Fortran::runtime;
 
+/// Generate call to Matmul intrinsic runtime routine.
+mlir::Value Fortran::lower::genMatmul(Fortran::lower::FirOpBuilder &builder,
+                                      mlir::Location loc, mlir::Value resultBox,
+                                      mlir::Value matrixABox,
+                                      mlir::Value matrixBBox) {
+  mlir::FuncOp func;
+  auto ty = matrixABox.getType();
+  auto arrTy = fir::dyn_cast_ptrOrBoxEleTy(ty);
+
+  func = Fortran::lower::getRuntimeFunc<mkRTKey(Matmul)>(loc, builder);
+  auto fTy = func.getType();
+  auto sourceFile = Fortran::lower::locationToFilename(builder, loc);
+  auto sourceLine =
+      Fortran::lower::locationToLineNo(builder, loc, fTy.getInput(4));
+  auto args =
+      Fortran::lower::createArguments(builder, loc, fTy, resultBox, matrixABox,
+                                      matrixBBox, sourceFile, sourceLine);
+  return builder.create<fir::CallOp>(loc, func, args).getResult(0);
```

Comment:
Shouldn't `genMatmul` return `void` ? `Matmul` runtime functions does not return anything.

Reply:
Yeah. :)

Reply:
Done.